mmcls/core/evaluation/mean_ap.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import torch
+
+
+def average_precision(pred, target):
+    """ Calculate the average precision for a single class
+
+    Args:
+        pred (np.ndarray): The model prediction with shape (N, C), where C is
+            the number of classes.
+        target (np.ndarray): The target of each prediction with shape (N, C),
+            where C is the number of classes.
+
+    Returns:
+        float: a single float as average precision value.
+    """
+    eps = np.finfo(np.float32).eps
+
+    # sort examples
+    sort_inds = np.argsort(-pred)
+    sort_target = target[sort_inds]
+
+    # count true positive examples
+    p_inds = sort_target == 1
+    tp = np.cumsum(p_inds)
+    total_p = tp[-1]
+
+    # count not difficult examples
+    pn_inds = sort_target != -1
```

Comment:
add an option for ignoring difficult samples, instead of hard code.

Reply:
Could you please be more specific about this comment? The code now can calculate map whether there are difficult examples. Thanks.